### PR TITLE
Add support for the PHP League CommonMark Engine

### PR DESCRIPTION
**Pull request by caseyamcl**
I suggest including support for the new PHP League CommonMark engine.  I'm using that in several projects, and think that others may adopt it too.

### CODE REVIEW

File: src/Aptoma/Twig/Extension/MarkdownEngine/PHPLeagueCommonMarkEngine.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: casey
+ * Date: 1/5/15
+ * Time: 4:54 PM
+ */
```
Comment (gunnarlium) [inline src/Aptoma/Twig/Extension/MarkdownEngine/PHPLeagueCommonMarkEngine.php]: No need for this. Git keeps track of when files where created, and who made them.

File: tests/Aptoma/Twig/Extension/MarkdownEngine/PHPLeagueCommonMarkEngineTest.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: casey
+ * Date: 1/5/15
+ * Time: 4:59 PM
+ */
```
Comment (gunnarlium) [inline tests/Aptoma/Twig/Extension/MarkdownEngine/PHPLeagueCommonMarkEngineTest.php]: Same as above, remove.

File: tests/Aptoma/Twig/Extension/MarkdownEngine/PHPLeagueCommonMarkEngineTest.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace Aptoma\Twig\Extension\MarkdownEngine;
+
+use Aptoma\Twig\Extension\MarkdownExtensionTest;
+
+// Require parent class if not autoloaded
+if ( ! class_exists('\Aptoma\Twig\Extension\MarkdownExtensionTest')) {
```
Comment (gunnarlium) [inline tests/Aptoma/Twig/Extension/MarkdownEngine/PHPLeagueCommonMarkEngineTest.php]: There should not be a space on either side of the `!`:

````PHP
if (!class_exists('\Aptoma\Twig\Extension\MarkdownExtensionTest')) {
    require_once(__DIR__ . '/../MarkdownExtensionTest.php');
}
````

**Comment by gunnarlium** [comment]
Hi!

Thanks for your contribution. I've commented on few minor style changes. Also make sure that every file has a new line as the last character. As you're using PhpStorm, I recommend enabling "Ensure line feed at file end on Save" in the options under Editor -> General -> Other (at the bottom).

I you fix those small issues and squash this to one commit, I'll be happy to merge.

**Comment by caseyamcl** [comment]
Hey thanks for reviewing so quickly!

I believe I've made the changes you indicated.  I also squashed the request into a single commit.  Let me know if I can make any other changes.

PS> thanks for the tip re: PHPStorm; I keep meaning to turn that feature off anyway.

**Pull request merged**